htdocs/main.php

```diff
@@ -118,44 +118,6 @@ function tplFromRequest($param)
 
     //--------------------------------------------------
 }
-$paths = $config->getSetting('paths');
-
-if (!empty($TestName)) {
-    $base = $paths['base'];
-
-    if (!empty($_REQUEST['sessionID'])) {
-        try {
-            $timePoint = TimePoint::singleton($_REQUEST['sessionID']);
-        } catch (Exception $e) {
-            $tpl_data['error_message'][] = htmlspecialchars($e->getMessage());
-        }
-    }
-
-    $paths = $config->getSetting('paths');
-
-    // get candidate data
-    if (!empty($_REQUEST['candID'])) {
-        try {
-            $candidate =& Candidate::singleton($_REQUEST['candID']);
-            $tpl_data['candidate'] = $candidate->getData();
-        } catch(Exception $e) {
-            $tpl_data['error_message'][] = htmlspecialchars($e->getMessage());
-        }
-    }
-
-    // get time point data
-    if (!empty($_REQUEST['sessionID'])) {
-        try {
-            $timePoint =& TimePoint::singleton($_REQUEST['sessionID']);
-            if ($config->getSetting("SupplementalSessionStatus")) {
-                $tpl_data['SupplementalSessionStatuses'] = true;
-            }
-            $tpl_data['timePoint'] = $timePoint->getData();
-        } catch(Exception $e) {
-            $tpl_data['error_message'][] = htmlspecialchars($e->getMessage());
-        }
-    }
-}
```

Comment:
@driusan fun exercise:

Count the number of redundant declarations/code duplication in these 30 lines of code

Reply:
0, the lines are deleted?